File: app/src/main/java/com/example/p2invest/view/MainActivity.java
package com.example.p2invest.view;

import android.view.WindowManager;
import android.widget.RadioGroup;

import androidx.fragment.app.FragmentTransaction;

import com.example.base.BaseActivity;
import com.example.p2invest.R;
import com.example.p2invest.fragments.HomeFragment;
import com.example.p2invest.fragments.MoreFragment;
import com.example.p2invest.fragments.TouziFragment;
import com.example.p2invest.fragments.WodeFragment;
import com.gyf.immersionbar.ImmersionBar;

public class MainActivity extends BaseActivity {

    private RadioGroup gr;

    @Override
    protected void initListener() {
       // ImmersionBar.with(this).init();

        gr.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup radioGroup, int i) {

                if (i== R.id.rb1){
                    getSupportFragmentManager().beginTransaction().replace(R.id.mainff,new HomeFragment()).commit();
                }
                else   if (i==R.id.rb2){
                    getSupportFragmentManager().beginTransaction().replace(R.id.mainff,new TouziFragment()).commit();
                }
                else    if (i==R.id.rb3){
                    getSupportFragmentManager().beginTransaction().replace(R.id.mainff,new WodeFragment()).commit();
                }
                else    if (i==R.id.rb4){
                    getSupportFragmentManager().beginTransaction().replace(R.id.mainff,new MoreFragment()).commit();
                }

            }


        });
    }
    @Override
    protected void initdata() {
        getSupportFragmentManager().beginTransaction().replace(R.id.mainff,new HomeFragment()).commit();
        initView();
    }

    @Override
    protected void initview() {

    }

    @Override
    protected int getlayout() {
        return R.layout.activity_main;
    }


    private void initView() {
        gr = (RadioGroup) findViewById(R.id.gr);

    }
}

File: app/src/main/java/com/example/p2pmonthhomework/MainActivity.kt
package com.example.p2pmonthhomework

import android.widget.RadioGroup
import androidx.fragment.app.FragmentManager
import com.example.base.BaseActivity
import com.example.p2pmonthhomework.fragments.FragmentHome
import com.example.p2pmonthhomework.fragments.FragmentInvestment
import com.example.p2pmonthhomework.fragments.FragmentMore
import com.example.p2pmonthhomework.fragments.FragmentProperty
import kotlinx.android.synthetic.main.main_bottom.*

class MainActivity : BaseActivity() {

    private lateinit var manager: FragmentManager
    lateinit var fragmenthome: FragmentHome
    lateinit var fragmentinvestment: FragmentInvestment
    lateinit var fragmentproperty: FragmentProperty
    lateinit var fragmentmore: FragmentMore

    override fun initData() {
        initFragments()
        initFoot()
        setHome()
    }

    override fun initView() {

    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }


    private fun initFoot() {
        rg_group.setOnCheckedChangeListener(object : RadioGroup.OnCheckedChangeListener {
            override fun onCheckedChanged(p0: RadioGroup?, p1: Int) {
                when (p1) {
                    R.id.rb_main_home -> setHome()
                    R.id.rb_main_invest -> setInvest()
                    R.id.rb_main_me -> setMe()
                    R.id.rb_main_more -> setMore()
                }
            }
        })
    }


    private fun setMore() {
        setTextBlack()
        rb_main_more.setTextColor(resources.getColor(R.color.home_back_selected))
        val transaction = manager.beginTransaction()
        transaction.replace(R.id.ll_main, fragmentmore)
        transaction.commit()
    }

    private fun setMe() {
        setTextBlack()
        rb_main_me.setTextColor(resources.getColor(R.color.home_back_selected01))
        val transaction = manager.beginTransaction()
        transaction.replace(R.id.ll_main, fragmentproperty)
        transaction.commit()
    }

    private fun setInvest() {
        setTextBlack()
        rb_main_invest.setTextColor(resources.getColor(R.color.home_back_selected))
        val transaction = manager.beginTransaction()
        transaction.replace(R.id.ll_main, fragmentinvestment)
        transaction.commit()
    }

    private fun setHome() {
        setTextBlack()
        rb_main_home.setTextColor(resources.getColor(R.color.home_back_selected))
        val transaction = manager.beginTransaction()
        transaction.replace(R.id.ll_main, fragmenthome)
        transaction.commit()
    }

    fun setTextBlack() {
        rb_main_home.setTextColor(resources.getColor(R.color.home_back_unselected))
        rb_main_invest.setTextColor(resources.getColor(R.color.home_back_unselected))
        rb_main_me.setTextColor(resources.getColor(R.color.home_back_unselected))
        rb_main_more.setTextColor(resources.getColor(R.color.home_back_unselected))
    }

    private fun initFragments() {
        manager = supportFragmentManager
        val transaction = manager.beginTransaction()

        fragmenthome = FragmentHome()
        fragmentinvestment = FragmentInvestment()
        fragmentproperty = FragmentProperty()
        fragmentmore = FragmentMore()

        transaction.add(R.id.ll_main, fragmenthome)
        transaction.add(R.id.ll_main, fragmentinvestment)
        transaction.add(R.id.ll_main, fragmentproperty)
        transaction.add(R.id.ll_main, fragmentmore)

        transaction.commit()
    }
}